### PR TITLE
Add Portal container logic

### PR DESCRIPTION
**Pull request by w-gao**
Problem
=======
Portal container logic [Jira](https://cruzhacks-dev.atlassian.net/browse/C2D-117)

Solution
========
What I/we did to solve this problem
* Added
  * function to fetch application status from the API (`application/checkApp` endpoint)
  * logic to render different components based on response

Change Summary:
---------------
* Tidy, well formulated commit message
* Another great commit message
* Something else I/we did

Dev-Ops
=======
If you added an ENV variable, please check off that you've updated the following  
- [ ] Key & Sample value to `.env` & `sample.env` 
- [ ] GCP Secret Manager (Both development and production)
- [ ] Github Secrets (Added a development and production variable)
- [ ] Github Workflows `.github/workflows/dev.yml` & `.github/workflows/prod.yml`
- [ ] webpack-config.js

Images/Important Notes (optional):
-----------------------
* Insert any notes/issues, dependencies added or removed, or images

### CODE REVIEW

File: src/views/Portal/index.view.tsx [hunk 1]
```diff
@@ -1,8 +1,65 @@
...
+              break
+            case "accepted":
+              setStatus(AppStatus.Accepted)
+              store("applicationStatus", AppStatus.Accepted, 12 * 60 * 60) //  store for 12hrs
+              break
+            case "rejected":
+              setStatus(AppStatus.Rejected)
+              store("applicationStatus", AppStatus.Rejected, 12 * 60 * 60) // store for 12hrs
+              break
+            case "pending":
```
Comment (krishnakpandian) [inline src/views/Portal/index.view.tsx:45]: It might be okay to store this as well since once they submit it won't change until after release

**Review by krishnakpandian — APPROVED** [review]
LGTM!

File: src/views/Portal/index.view.tsx (hunk 1 above)
Comment (w-gao) [reply src/views/Portal/index.view.tsx:45]: that makes sense

**Comment by w-gao** [comment]
Merging this in but we can make additional changes as the design updates.

**Pull request merged**